### DISCUSSION TITLE
OpenHantek Compatibility for DSO2C10, DSO2C15, DSO2D10, DSO2D15

### DISCUSSION THREAD

**Discussion by bzettler59** [OpenHantek/OpenHantek6022#320] [Q&A] [answered]
These scopes are at a price point where many embedded engineers will want them, but OpenHantek doesn't recognize my 2D15. For some odd reason, it comes up as a Compaq USB device (0x049F:505E). I'm running Linux.

I'm sure they're all essentially identical from an interfacing standpoint.

Are there plans to support these scopes?

**Comment by Ho-Ro** [comment, marked as answer]
> Are there plans to support these scopes?

Short answer: No.
Reason: I do not support devices that I do not have on my desk. OH6022 is written to support my DSO-6022 (and some more or less similar devices) and is optimized for the underlying USB protocol. As long as even sigrok has no support for these devices - no chance.

> I'm sure they're all essentially identical from an interfacing standpoint.

And if you're _sure_ it is a no-brainer, feel free to fork OH6022 and write OH2D15 or provide a working PR.

**Reply by bzettler59** [reply]
Thanks so much for responding. I also have a 6022 USB scope, so I've used your excellent software before.

I meant all the scopes in this family would be nearly identical to *each other*, not any of the other Hantek scope families. I'm sure they're different from those scopes, and the fact that they come up with a Compaq VendorId is not a good sign. It means they're a very sloppy software shop and let something out the door and into the marketplace that will needlessly annoy something (Hewlett-Packard) who has many, many lawyers.

Adding to the confusion is that Hantek has two entirely different lines of scopes called DSO2000. The *other* DSO2000 scopes are supported by OpenHantek, but these are not. Oh, well.

**Comment by Ho-Ro** [comment]
> they come up with a Compaq VendorId

Also the 6022 announce their VendorId 0x04b4 = Cypress Semiconductor (the producer of the EzUSB chip that does the complete work of the scope) before FW upload. When the FW is loaded VID changes to 0x04b5 = ROHM LSI Systems USA, LLC - this is an absolute nonsense.

There is more info about the protocol of other Hanteks: [German](https://www.mikrocontroller.net/articles/Hantek_Protokoll) and [English](https://elinux.org/Das_Oszi_Protocol).
You could also try the legacy project [openhantek](https://github.com/OpenHantek/openhantek) that supports some more Hantek scopes.
